README.md

```diff
@@ -66,47 +66,34 @@ struct ExampleStorageView: View {
+The `CredentialsStorage` module allows for the encrypted storage of small chunks of sensitive user data, such as usernames and passwords for internet services, using Apple's [Keychain documentation](https://developer.apple.com/documentation/security/keychain_services/keychain_items/using_the_keychain_to_manage_user_secrets). 
 
 Credentials can be stored in the Secure Enclave (if available) or the Keychain. Credentials stored in the Keychain can be made synchronizable between different instances of user devices.
 
 ### Handling Credentials
 
-Use the `SecureStorage` module to store a set of [`Credentials`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezisecurestorage/credentials) instances in the Keychain associated with a server that is synchronizable between different devices.
+Use the `CredentialsStorage` module to store a set of [`Credentials`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezicredentialsstorage/credentials) instances in the Keychain associated with a server that is synchronizable between different devices.
 
-- Store new credentials: [`store(credentials:server:removeDuplicate:storageScope:)`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezisecurestorage/securestorage/store(credentials:server:removeduplicate:storagescope:))
-- Retrieve existing credentials: [`retrieveCredentials(_:server:accessGroup:)`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezisecurestorage/securestorage/retrievecredentials(_:server:accessgroup:))
-- Retrieve all matching existing credentials: [`retrieveAllCredentials(forServer:accessGroup:)`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezisecurestorage/securestorage/retrieveallcredentials(forserver:accessgroup:))
-- Update existing credentials: [`updateCredentials(_:server:newCredentials:newServer:removeDuplicate:storageScope:)`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezisecurestorage/securestorage/updatecredentials(_:server:newcredentials:newserver:removeduplicate:storagescope:))
-- Delete existing credentials: [`deleteCredentials(_:server:accessGroup:)`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezisecurestorage/securestorage/deletecredentials(_:server:accessgroup:))
-- Delete all matching existing credentials: [`deleteAllCredentials(itemTypes:accessGroup:)`](https://swiftpackageindex.com/stanfordspezi/spezistorage/documentation/spezisecurestorage/securestorage/deleteallcredentials(itemtypes:accessgroup:))
 
 
 ### Handling Keys
 
-Similar to `Credentials` instances, you can also use the `SecureStorage` module to interact with keys.
+Similar to `Credentials` instances, you can also use the `CredentialsStorage` module to interact with cryptographic keys.
```

Comment:
I am wondering if we should introduce an abstraction layer for the keychain interactions (just internal to the package, e.g. exposed with `package`) and then have a `CredentialsStorage` and `KeyStorage` that are separate targets & types?

Reply:
i really like this idea. it'd also allow us to easily add the ability to allow apps to store certificates using `CredentialsStorage` (since these are just another form of keychain item).

we might even want to make the keychain abstraction layer public[1] in case apps need that level of raw access / need to go beyond the level of control the CredentialsStorage provides.

[1] similar to like how with SpeziBluetooth and SpeziDevices we both have a package that implements the low-level BT interactions, plus one that give you a couple of well-known device interactions and UI, if that's sufficient for your needs

Reply:
Agree; that would be great! And yes, it might even make sense to make that public as well, good point.